awx/api/views/__init__.py

```diff
@@ -1542,7 +1530,13 @@ class InventoryHostsList(HostRelatedSearchMixin, SubListCreateAttachDetachAPIVie
 
     def get_queryset(self):
         inventory = self.get_parent_object()
-        return getattrd(inventory, self.relationship).all()
+        qs = getattrd(inventory, self.relationship).all()
+        # Apply queryset optimizations
+        if HostAccess.select_related:
+            qs = qs.select_related(*HostAccess.select_related)
+        if HostAccess.prefetch_related:
+            qs = qs.prefetch_related(*HostAccess.prefetch_related)
+        return qs
```

Comment:
Ever since smart inventories were introduced `/api/v2/inventories/N/hosts/` was extremely un-optimized, which is unfortunate since it's a highly trafficked endpoint. The particular properties of this sublist (coming from the host_filter) prevent us from sourcing the queryset from the `HostAccess` queryset directly, but that's no reason to not use the optimizations from it.

Reply:
Why have the conditionals here?  `select_related` and `prefetch_related` are defined on `HostAccess`.

Reply:
yeah, that's true. It's purely static content, only imported to be DRY